vendors/microchip/boards/ecc608a_plus_winsim/aws_demos/application_code/aws_demo_logging.c

```diff
@@ -321,20 +325,25 @@ static void prvLoggingPrintf( BaseType_t xFormatted,
             pcTaskName = pcNoTask;
         }
 
-        if( ( strcmp( pcFormat, "\n" ) != 0 ) && ( xFormatted != pdFALSE ) )
+        if( ( xAfterLineBreak == pdTRUE ) &&
+            ( strcmp( pcFormat, "\r\n" ) != 0 ) &&
+            ( xFormatted != pdFALSE ) )
         {
-            xLength = snprintf( cPrintString, 
-                                dlMAX_PRINT_STRING_LENGTH, 
+            xLength = snprintf( cPrintString,
+                                dlMAX_PRINT_STRING_LENGTH,
                                 "%lu %lu [%s] ",
                                 xMessageNumber++,
-                                ( unsigned long )xTaskGetTickCount(),
+                                ( unsigned long ) xTaskGetTickCount(),
                                 pcTaskName );
+            xAfterLineBreak = pdFALSE;
         }
         else
```

Comment:
Should this be updated to make sure that the `xAfterLineBreak` flag is set only after a line break message?
```suggestion
        else  if( strcmp( pcFormat, "\r\n" ) == 0 )
```

Reply:
That won't work though. We only want to clear the buffer if the previous log message was a line break rather than the current log message.